### PR TITLE
Added support for online synced folder path

### LINKED ISSUE #351: use a folder/file instead of a Gist
Would be nice if a folder or file could be used for syncing instead of a Gist. Would work for SMB, NFS, Dropbox, Google Drive, etc.

### PR DESCRIPTION
**Pull request by pksunkara**
Fixes #351

### CODE REVIEW

**Comment by dirk-thomas** [comment]
In the context a various different backends to store the actual data being requested I would rather like to see a more scalable approach. The current patch adds various conditions in multiple places. If more backends would be added in the future that seem to likely become unmanageable. Does that make any sense?

---

Since this package is only "sparsely" maintained it is even more important to have a good test coverage to avoid breakage of functionality. Please add test coverage for the newly added feature in order for the PR to be considered for being merged. This can "wait" though until the patch is "feature complete" in order to not waste effort on writing specs when the patch isn't ready to be considered for merging yet.

**Comment by pksunkara** [comment]
@dirk-thomas When you say `patch is "feature complete" in order`, you want me to make the backend scalable, right? Or was there anything else? I am going to try to get this ready to be merged.

**Comment by dirk-thomas** [comment]
@pksunkara Since a few months ago I have stopped using Atom in favor of a different editor. Hence I won't spend my spare time on this package anymore. Hopefully one of the other users maintaining the package in the past will be able to help. Sorry.

**Comment by pksunkara** [comment]
The PR code is now ready to be reviewed:

1. Made a working plugin system
2. Move gist logic to gist plugin
3. Added folder plugin
4. Added specs for the folder plugin

@eimfach Could you review it too?

**Comment by pksunkara** [comment]
@dirk-thomas I understand. Thanks for all the work. If no one's interested in maintaining this, I would be willing to do it. Would you mind reviewing this PR, and once this is merged, make me a colloborator?

**Comment by dirk-thomas** [comment]
@jerone @groupsky @joefitzgerald @adamrenklint Since all of you have been maintaining the package in the past or have collaboration roles for this repo I wanted to get your opinion too. I don't mind granting access to a potential new maintainer since I won't have the time and interest to fulfill the role anymore.

**Comment by dirk-thomas** [comment]
@pksunkara Currently CI is failing for your patch. You might want to look into why that is the case and get it "green" in the meantime.

**Comment by pksunkara** [comment]
Fixed the tests. Just a small issue. PR's done and ready

**Comment by pksunkara** [comment]
@dirk-thomas Since they are not interested, I think we should move forward.

**Comment by eimfach** [comment]
@pksunkara Will have a look at it ! 
@dirk-thomas I am also interested to maintain this repo

**Comment by jerone** [comment]
@dirk-thomas I have no opinion against granting others access to this package to keep this package alive. Go for it.

File: lib/plugins/folder.coffee [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  getFolderPath: ->
+    folderPath = atom.config.get('sync-settings.folderPath')
+    if folderPath
+      folderPath = folderPath.trim()
+    return folderPath
+
+  blacklistKeys: (blacklistedKeys) ->
+    blacklistedKeys.push('sync-settings.folderPath')
+
+  initialize: ->
```
[the remark this answers is not in the document]
Comment (pksunkara) [reply lib/plugins/folder.coffee]: It's part of the plugin API. Check the gist plugin.

File: lib/plugins/folder.coffee [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+  checkMissing: (missingSettings) ->
+    if not @getFolderPath()
+      missingSettings.push("Folder Path to sync")
+
+  check: (cb) ->
+    if not @getFolderPath()
+      return notifyMissingMandatorySettings(["Folder Path to sync"])
+
+    console.debug('checking latest backup from synced folder...')
```
[the remark this answers is not in the document]
Comment (pksunkara) [reply lib/plugins/folder.coffee]: They were there from before

File: lib/plugins/folder.coffee (hunk 1 above)
[the remark this answers is not in the document]
Comment (eimfach) [reply lib/plugins/folder.coffee]: alright

**Comment by eimfach** [comment]
@pksunkara 
The PR you provide has a lot of changes, also I see that you retouched/rearranged existing code but nothing changed there, which blurs the history of those code lines  (and I don't understand why you did that).
Some explanation from your side e.g. a PR Summary would be very helpful..

It seems a lot to me just for the sake of using a file from fs instead from a url.

**Pull request closed without merging**